test/connectors/test_cpu_torch_connector.py

```diff
@@ -0,0 +1,37 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""CPU based tests of the PyTorch connector."""
+
+import unittest
+from test import QiskitMachineLearningTestCase
+from test.connectors.test_torch_connector import TestTorchConnector
+
+from qiskit.utils import optionals
+import qiskit_machine_learning.optionals as _optionals
+
+
+class TestCPUTorchConnector(QiskitMachineLearningTestCase, TestTorchConnector):
+    """CPU based tests of the PyTorch connector."""
+
+    @unittest.skipUnless(optionals.HAS_AER, "qiskit-aer is required to run this test")
+    @unittest.skipIf(not _optionals.HAS_TORCH, "PyTorch not available.")
```

Comment:
Just curious why one uses `skipUnless` and the other `skipIf(not ` - it just seems odd since I think they end up doing the same right (same in the similar class for GPU)

Reply:
Yes, it is just because the AER test was copied form other files in ML and were initially copied from Terra that uses `skipunless` for Aer and uses `skipIf` for other optional libraries but they do the same.

Reply:
I get they do the same - and maybe in one library in different places it would seem a lot less weird than when the occurrences are back-to-back like this. It makes me wonder whether there is some reason for it since while they should work the same way why are the two different forms used here by the author. As in why is the PyTorch one not simply a skipUnless too.

Reply:
I changed to skipUnless and refactored to a common base class.